src/bin/client-demo.rs

```diff
@@ -395,6 +395,8 @@ fn converge(
 }
 
 fn read_leader(path: String) -> ReplicatedData {
-    let file = File::open(path).expect("file");
-    serde_json::from_reader(file).expect("parse")
+    let file = File::open(path.clone())
```

Comment:
looks like someone's been reading their rust book

Reply:
Yeah, `let file = File::open(&path)` works too.  I reached for that first ironically, but then I saw https://github.com/solana-labs/solana/blob/master/src/bin/fullnode.rs#L114 and cargo cult in the `.clone()`.

Reply:
understanding borrow is my #1 rust Q right now